packages/registry/db/seeds/namespaceSeeder.ts

```diff
@@ -2,103 +2,292 @@ import { homedir } from 'os';
+
+      parentRotation = (await rotationRepo.find({ where: { serviceId: foundParent.id }, order: { serviceRotation: 'desc' }, take: 1 }))[0]
+        .serviceRotation;
+      parent = { ...foundParent, rotation: parentRotation };
+    }
+
+    const savedService = await serviceRepo.save({
+      namespaceId: namespace.namespaceId,
+      name: service.name,
+      parallelism: service.parallelism,
+      serviceType: service.serviceType,
+      parentServiceId: parent?.parentServiceId ?? null,
+      parent,
+    });
+
+    await rotationRepo.save({
+      service: savedService,
+      serviceId: savedService.id,
+      serviceRotation: service.rotation ?? INITAL_ROTATION_VALUE,
+      parentRotation,
+    });
+
+    console.log(`created service ${service.name}`);
+  }
+
+  const modifications = seedInput.services?.filter((service) => service.action === 'modify') as ServiceModifyInput[];
+  for await (const service of modifications) {
+    if (service.parallelism === undefined && service.serviceType === undefined) {
+      continue;
+    }
+
+    const fectchedService = await serviceRepo.findOneOrFail({ where: { namespaceId: namespace.namespaceId, name: service.name } });
+    await serviceRepo.update(fectchedService.id, {
+      parallelism: service.parallelism ?? fectchedService.parallelism,
+      serviceType: service.serviceType ?? fectchedService.serviceType,
+    });
+
+    console.log(`modified service ${service.name}`);
+  }
+
+  // create blocks for newly created and modified services
+  for await (const service of [...creations, ...modifications]) {
+    const blockerService = await serviceRepo.findOneOrFail({ where: { namespaceId: namespace.namespaceId, name: service.name } });
+
+    for (const blockee of service.blockees ?? []) {
+      const blockeeService = await serviceRepo.findOneOrFail({ where: { namespaceId: namespace.namespaceId, name: blockee } });
+
+      const existingBlock = await blockRepo.findOne({ where: { blockerId: blockerService.id, blockeeId: blockeeService.id } });
+
+      if (existingBlock === null) {
+        await blockRepo.save({
+          blockerId: blockerService.id,
           blockerService: blockerService,
-          blockeeId: (blockeeService as Service).id,
+          blockeeId: blockeeService.id,
           blockeeService: blockeeService,
         });
+
+        console.log(`created block ${blockerService.name} on ${blockeeService.name}`);
       }
     }
+  }
+
+  return { id: namespace.namespaceId, name: seedInput.name, services: new Map() };
+};
+
+export interface NamespaceSeederOutput {
+  id: number;
+  name: string;
+  services: Map<string, Service & { rotation: number }>;
+}
+
+export const SEEDER_INPUT_PATH = join(homedir(), SEEDER_INPUT_FILE_NAME);
 
-    return { id: namespaceId, name: seedInput.name, services: map };
+export default class NamespaceSeeder implements Seeder {
+  public async run(dataSource: DataSource, factoryManager: SeederFactoryManager): Promise<NamespaceSeederOutput> {
```

Comment:
eslint warning

Reply:
at the moment I've added `eslint-disable-next-line`,
the use of `typeorm-extension` for seeding is a bit of a work around. it was more relevant for the previous iteration where we just seeded data. but this pr gives as an optional modification and deletion of existing data.
I've kept the use of `typeorm-extension` for the `seed` command line options while not using it's `SeederFactoryManager` class because it doesn't support transactions and database updates.

in the future namespace modification functionality should be behind an api on `registry`, but at this moment this workaround gives us the option to modify a namespace safely.